procrastinate/worker.py

```diff
@@ -338,6 +335,39 @@ async def run(self):
                 pass
             raise
 
+    async def _handle_notification(
+        self, *, channel: str, notification: jobs.Notification
+    ):
+        if (
+            notification["type"] == "job_inserted"
+            and notification["status"] == jobs.Status.TODO.value
+        ):
+            self._new_job_event.set()
+        elif notification["type"] == "abort_job_requested":
+            self._job_ids_to_abort.add(notification["job_id"])
```

Comment:
It seems that aborted IDs only accumulate in `_job_ids_to_abort`. Should we remove aborted job IDs from this set or clean it up somehow differently?

Reply:
I had the same thought. 

At the time, I brushed it away deciding that it will probably not be a real problem in practice.

That said, it should be straightforward to attempt to remove any job id that completed from that list. 

That should only leave ids in the set in certain circumstances (such as multiple worker processes sharing the same queue(s).)